internal/config/config.go

```diff
@@ -0,0 +1,140 @@
+package config
+
+import (
+	"fmt"
+	"os"
+	"sort"
+
+	"sigs.k8s.io/yaml"
+)
+
+type distributionName = string
+type claimName = string
+type Distributions map[distributionName]Distribution
+type Entitlements map[claimName][]distributionName
+
+type Distribution struct {
+	ID     string `json:"id"`
+	Prefix string `json:"prefix"`
+}
+
+// for marshalling purposes
+type config struct {
+	Distributions Distributions `json:"distributions"`
+	Entitlements  Entitlements  `json:"entitlements"`
+}
```

Comment:
This seems like the server config struct. It should own loading, parsing, and validating of configs from file. That would be tooled into the server startup for initial loading and reloading on a file watch. 
1. Load, parse validate
2. Update whatever the Config struct becomes.

Reply:
This seems more like an anonymous struct used purely to hydrate Config from a yaml file.

Reply:
my intention was to have a private struct to hydrate Config as David mentioned. I also saw a pattern of setting an anonymous struct in parse, for example.